Controller/Process/Result.php

```diff
@@ -389,7 +389,21 @@ protected function validatePayloadAndReturnResponse($response)
             $request['paymentData'] = $this->_session->getLastRealOrder()->getPayment()->getAdditionalInformation("paymentData");
         }
 
-		$request["details"] = $response;
+        if (!empty($this->_session->getLastRealOrder()) &&
+            !empty($this->_session->getLastRealOrder()->getPayment()) &&
+            !empty($this->_session->getLastRealOrder()->getPayment()->getAdditionalInformation("details"))
+        ) {
+            $details = $this->_session->getLastRealOrder()->getPayment()->getAdditionalInformation("details");
+            $key = array_search('returnUrlQueryString', $details[0]);
+
+            if ($key !== false) {
+                $request['details']['returnUrlQueryString'] = http_build_query($response);
+            } else {
+                $request["details"] = $response;
+            }
+        } else {
+            $request["details"] = $response;
+        }
```

Comment:
a bit concerned about this flow, is this something our merchants need ?

Reply:
Yes unfortunately for checkout directIssuerPages we need to implement this extra step - so currently this is a quick fix for those who use this setting but in the future this account setting can become default and can be that the API will also adjust.
For now I would just use it like this and let's follow up with the checkout team if they are going to make it more generic later.

Reply:
We can declare $request["details"] = $response; first and then override it with the if statement. That results in 2 else statements we can remove and cleaner code. What do you think ?